build.gradle

```diff
@@ -36,11 +36,13 @@ repositories {
 }
 
 dependencies {
-    compile 'junit:junit:4.12'
+    compile 'org.assertj:assertj-core:3.8.0'

```

Comment:
As far as I recall, AssertJ 3.x targets Java 8, can it be used with Android? Although current Android toolchain generally supports Java 8, the best practice is to compile libraries against Java 6 if you're targeting Android.

Reply:
Yes, AssertJ targets Java8. Nicely spotted @Egorand 
I guess this is no real option to just drop the support for the whole Android World.

Reply:
Good call on the Android dependency. I was unaware of the Java 6 limitation.

We could move to [AssertJ 1.x for Java 6](https://joel-costigliola.github.io/assertj/assertj-core.html) though Core seems to suggest that you could use 2.x/3.x with Java6Assertions as the entry point. Honestly, we're not using any of the advanced features of AssertJ and moving to 1.x wouldn't be the worst thing.

Reply:
You should just aim for a lib (in that case for a branch), which is still maintained. Or at least accepts pull requests for that branch. With assertJ, it not seems that the 1.x branch is development. I'm not sure if PR's to the 1.x would actually make it to the repo and to a new version of the 1.x stream.
That's quite sad, because I also really like AssertJ but IMO branch 2.x is no option (because of java 8) and 1.x seams pretty dead.